crates/sui-core/src/checkpoints/checkpoint_executor/mod.rs

```diff
@@ -57,10 +58,90 @@ type CheckpointExecutionBuffer = FuturesOrdered<
     )>,
 >;
 
+#[derive(Clone)]
+pub struct EndOfEpochMessage {
```

Comment:
Why not just use Committee?

Reply:
The reconfig code you added assumes we receive the new epoch number, so figured I would return both

Reply:
The Committee data type is pretty much the same as EndOfEpochMessage

Reply:
The `next_epoch` is in there as well.... I could have returned a tuple as well. The struct vs tuple was just to make pattern matching a bit more readable.

Reply:
I am referring to this: https://github.com/MystenLabs/sui/blob/7d4c2aaf474da33adcd348e0f301c866e832d68c/crates/sui-types/src/committee.rs#L37
We could have used this type all the way along.
The primary reason we are not storing `Committee` type directly in checkpoint summary is that it contains somewhat redundant information (e.g. epoch number). When we take it out of checkpoint summary, if we decide to include epoch number, we might as well just create a `Committee` instance for simplicity

Reply:
Also, technically the reconfig channel only needs epoch number, not even committee, but that's a minor detail

Reply:
Ah I see. Thanks, will just use this

Reply:
@lxfind just now looking at this data structure. Not sure if we're able to materialize this struct from end of epoch in checkpoint executor as the only thing that gets included in the last checkpoint is the voting rights

Reply:
You already derive the next epoch id, so epoch + voting rights is basically Committee.

Reply:
Ah I didn't notice the constructor only took those. Sg